src/foam/nanos/approval/ApprovalDAO.js

```diff
@@ -108,8 +108,9 @@ foam.CLASS({
           initiatingUser = (User) userDAO.find(((ApprovalRequest) request).getCreatedFor());
         }
         subject.setUser(initiatingUser);
-        
-        X initiatingUserX = x.put("subject", subject);
+
+        X initiatingUserX = x.put("subject", subject)
+                             .put("ApprovalRequest", request);
```

Comment:
```suggestion
                             .put("approvalRequest", request);
```
Alternatively, if this is purely java side, then the following is also appropriate
`.put(ApprovalRequest.class, request) `
with the benefit that no casting is required on x.get()

Reply:
Updated to `.put(ApprovalRequest.class, request)`.